File: resources/[ui]/ps-ui/html/js/showImage.js
$(function () {
    function display(bool) {
        if (bool) {
            $("#showImage").show();
        } else {
            $("#showImage").hide();
        }
    }
    
    display(false)

    window.addEventListener('message', function(event) {
        var item = event.data;
        display(false)
        if (item.type === "showImage") {
            if (item.status == true) {
                display(true)
                document.getElementById("imgcontainer").innerHTML = '<img class="contain" style="width:800px;height:800px;" src="'+item.image+'" alt="'+item.image+'"></img>';
            } else {
                display(false)
            }
        }
    })

    document.onkeyup = function (data) {
        if (data.which == 27) {
            $.post(`https://ps-ui/showItemImage-callback`, JSON.stringify({}));
            return
        }
    };

    $("#showImageClose").click(function () {
        $.post(`https://ps-ui/showItemImage-callback`, JSON.stringify({}));
        return
    })
})

File: resources/[ui]/ps-ui/client/showImage.lua
local function SetDisplay(bool, img)
    SetNuiFocus(bool, bool)
    SendNUIMessage({
        type = "showImage",
        image = img,
        status = bool,
    })
end

local function ShowImage(img)
    SetDisplay(true, img)
end

RegisterNUICallback("showItemImage-callback", function(data, cb)
    SetDisplay(false)
    cb('ok')
end)

exports("ShowImage", ShowImage)